concepts/extensions/elasticsearch-concept.md

```diff
@@ -0,0 +1,52 @@
+# Elasticsearch
+
+Elasticsearch is a NoSQL Database focused on search capabilities to act as a search engine.
+The Shopware implementation of Elasticsearch provides an integrated way to improve the performance of product and category searches.
+To use Elasticsearch for your shop take a look at our [elasticsearch guide](guides/hosting/elasticsearch)
+
+## Concept
+
+### ElasticsearchDefinition
+To provide elasticsearch for an entity a corresponding `ElasticsearchDefinition` need to be added. Currently shopware has such a definition for the product entity called `ProductElasticsearchDefinition`.
+This definition defines the fields which are provided to elasticsearch and how they are aggregated.
+
+### ElasticsearchEntitySearcher
+The `ElasticsearchEntitySearcher` decorates the `EntitySearcher` to map the entity search to the elasticsearch structure.
+The `ElasticsearchEntitySearcher` return an `IdSearchResult` hydrated by the `ElasticsearchEntitySearchHydrator` as the `EntitySearcher` does and this result is used to read the found ids from the database.
+
+### ElasticsearchEntityAggregator
+The `ElasticsearchEntityAggregator` does the same as the `ElasticsearchEntitySearcher` for aggregations.
+
+### CriteriaParser
+The `CriteriaParser` parses the criteria to an elasticsearch specific notation.
+
+### ProductSearchBuilder
+The product search has a special `ProductSearchBuilder` in the core and so has the elasticsearch extension a corresponding extension for the `ProductSearchBuilder`.
+This extension matches the queries of the core `ProductSearchBuilder` to the elasticsearch notation.
+
+### ProductUpdater
+The `ProductUpdater` listens to the `ProductIndexerEvent` and triggers the `ElasticsearchIndexer` on changes to a `ProductEntity` 
+
+## Commands
+
+### es:index:cleanup
+The command `es:index:cleanup` deletes outdated elasticsearch indexes.
+The parameter -f will skip the confirmation.
+
+### es:create:alias
+The command `es:create:alias` refreshes the current elasticsearch index and sets the alias to the index name without the timestamp (which will make this index the active index).
```

Comment:
I would mention here that this step happens normally automatically. So this command can be used to force the alias creation